crates/tracing/src/metrics.rs

```diff
@@ -0,0 +1,51 @@
+use heck::ToKebabCase;
+
+#[cfg(feature = "otel")]
+const DEFAULT_METRICS_ENDPOINT: &str = "http://localhost:4318/v1/metrics";
+
+#[cfg(not(feature = "otel"))]
+pub fn configure_metrics(_: String, _: &wasmcloud_core::OtelConfig) -> anyhow::Result<()> {
+    // no-op for when otel isn't enabled.
+    Ok(())
+}
+
+#[cfg(feature = "otel")]
+#[allow(clippy::missing_errors_doc)]
+#[allow(clippy::module_name_repetitions)]
+pub fn configure_metrics(
+    service_name: &str,
+    otel_config: &wasmcloud_core::OtelConfig,
+) -> anyhow::Result<()> {
+    use opentelemetry_otlp::WithExportConfig;
+    let normalized_service_name = service_name.to_kebab_case();
+
+    let endpoint = if let Some(ref endpoint) = otel_config.exporter_otlp_endpoint {
+        endpoint.clone()
+    } else {
+        eprintln!(
+            "OTEL exporter endpoint not set, defaulting metrics to '{DEFAULT_METRICS_ENDPOINT}'"
+        );
+        DEFAULT_METRICS_ENDPOINT.to_string()
```

Comment:
I don't think we should default to this ourselves. Instead, we should let the underlying library handle whether to set a custom endpoint. I accomplished this in my PR for traces/logs by using a mutable builder: https://github.com/wasmCloud/wasmCloud/pull/1426/files#diff-4416842cf20f21d591f893e02fe590b6dd14be4ecdc21c1fc9879caec85a040bR226

Reply:
Agreed, there is plenty of defaulting to lean on from the library, using that is definitely the right move.